tests/test_data.py

```diff
@@ -500,3 +512,89 @@ def test_hap_gt_write(self):
 
         # remove the file
         os.remove(str(fname))
+
+
+class TestGenotypesRefAlt:
+    def test_read_ref_alt(self):
+        # simple.vcf
+        gts_ref_alt_read = GenotypesRefAlt(DATADIR.joinpath("simple.vcf"))
+        gts_ref_alt_read.read()
+        expected = np.array(
+            [
+                ("T", "C"),
+                ("A", "G"),
+                ("C", "A"),
+                ("A", "G"),
+            ],
+            dtype=gts_ref_alt_read.variants[["ref", "alt"]].dtype,
+        )
+        for i, x in enumerate(expected):
+            assert gts_ref_alt_read.variants[["ref", "alt"]][i] == x
+
+        # example.vcf.gz
+        gts_ref_alt = GenotypesRefAlt(DATADIR.joinpath("example.vcf.gz"))
+        gts_ref_alt.read()
+        expected = np.array(
+            [
+                ("C", "A"),
+                ("T", "C"),
+                ("G", "A"),
+                ("T", "C"),
+                ("A", "G"),
+                ("A", "G"),
+                ("T", "G"),
+                ("T", "A"),
+                ("G", "A"),
+                ("T", "C"),
+                ("C", "G"),
+                ("A", "G"),
+                ("T", "C"),
+                ("T", "C"),
+                ("C", "T"),
+            ],
+            dtype=gts_ref_alt.variants[["ref", "alt"]].dtype,
+        )
+        for i, x in enumerate(expected):
+            assert gts_ref_alt.variants[["ref", "alt"]][i] == x
+
+    def _get_fake_genotypes(self):
+        gts = Genotypes(fname=None)
+        gts.data = self._get_expected_genotypes()
+        gts.variants = np.array(
+            [
+                ("1:10114:T:C", "1", 10114, 0),
+                ("1:10116:A:G", "1", 10116, 0.6),
+                ("1:10117:C:A", "1", 10117, 0),
+                ("1:10122:A:G", "1", 10122, 0),
+            ],
+            dtype=[
+                ("id", "U50"),
+                ("chrom", "U10"),
+                ("pos", np.uint32),
+                ("aaf", np.float64),
+            ],
+        )
+        gts.samples = ("HG00096", "HG00097", "HG00099", "HG00100", "HG00101")
+        gts.check_phase()
+        return gts
+
+    def test_write_ref_alt(self):
+        gts_ref_alt_write = GenotypesRefAlt(DATADIR.joinpath("simple.vcf"))
+        gts_ref_alt_write.read()
```

Comment:
I think the main goal of this test is to verify that we can write the genotypes properly

so we should also write the genotypes to a temporary file and then read them again, so that we can verify that they still appear the same